src/ops/info.rs

```diff
@@ -23,14 +24,25 @@ pub fn info(
     spec: &PackageIdSpec,
     config: &Config,
     reg_or_index: Option<RegistryOrIndex>,
+    path: Option<&Path>,
 ) -> CargoResult<()> {
     let mut registry = PackageRegistry::new(config)?;
     // Make sure we get the lock before we download anything.
     let _lock = config.acquire_package_cache_lock(CacheLockMode::DownloadExclusive)?;
     registry.lock_patches();
 
+    // If the user specified a path, use it as a source.
+    let manifest_path = path.as_ref().map(|p| p.join("Cargo.toml"));
     // If we can find it in workspace, use it as a specific version.
-    let root_manifest = root_manifest(None, config).ok();
+    let root_manifest = root_manifest(manifest_path.as_deref(), config).ok();
+    if path.is_some() {
+        // We must normalize the path to ensure that it is a valid path.
+        let root = paths::normalize_path(&std::env::current_dir().unwrap().join(path.unwrap()));
+        let source_id = SourceId::for_path(&root)?;
+        let src = PathSource::new(&root, source_id, config);
+        registry.add_preloaded(Box::new(src));
+    }
```

Comment:
nit: for me, I feel like `if path.is_some()` would better fit when we do other registry operations

Reply:
Do you mean we should add it inside the `query_and_pretty_view` function? Adding it here would help us return earlier if the path is not a valid cargo workspace.

Reply:
I meant after the first line in the highlight, `registry.lock_patches();`

Reply:
Got it. Make sense.